astropy/io/fits/_tiled_compression/tiled_compression.py

```diff
@@ -260,24 +255,32 @@ def decompress_hdu(hdu):
     tile_shape = _tile_shape(hdu._header)
     data_shape = _data_shape(hdu._header)
 
-    data = np.zeros(data_shape, dtype=BITPIX2DTYPE[hdu._header["ZBITPIX"]])
+    first_array_index = first_tile_index * tile_shape
```

Comment:
Good catch, but today I learned that this works because:

```
In [2]: np.array([1, 2, 3]) * (2, 3, 4)
Out[2]: array([ 2,  6, 12])
```

I'm surprised, I thought this would have raised a TypeError - in any case I'll fix it here to explicitly make tile shape an array.

Reply:
Done

Reply:
Indeed, `ndarray` is surprisingly "greedy", converting anything it can...